### PR TITLE
Setuptools setup is required while build process

### PR DESCRIPTION
**Pull request by harshad16**
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

## Related Issues and Dependencies

#46 

## This introduces a breaking change

- [x] Yes
- [ ] No

This might have some effects on pytest and build process. 

## This Pull Request implements

Included setup content in `setup.py` for fixing the issue which occurs while the build operation `prepare_metadata_for_build_wheel`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -36,4 +41,40 @@ def run_tests(self):
...
+            # dirty, remove trailing and leading chars
+            return line.split(" = ")[1][1:-2]
+    raise ValueError("No version identifier found")
+
+
+VERSION = get_version()
+setup(
+    name="thoth-unresolved-package-handler",
+    version=VERSION,
+    description="Service to send/receive message regarding thoth solver.",
```
Comment (pacospace) [inline setup.py:68]: Service to send/receive message regarding failed advisers runs due to unresolved packages.

**Comment by goern** [comment]
here is a little something from pre-commit:
```
pydocstyle...............................................................Failed
- hook id: pydocstyle
- exit code: 1

setup.py:1 at module level:
        D100: Missing docstring in public module
setup.py:22 in public method `initialize_options`:
        D102: Missing docstring in public method
setup.py:26 in public method `finalize_options`:
        D102: Missing docstring in public method
setup.py:31 in public method `run_tests`:
        D102: Missing docstring in public method
```

**Comment by goern** [comment]
and ...
```
Tests should end in _test.py.............................................Failed
- hook id: name-tests-test
- exit code: 1

tests/test_unresolved_package_handler.py does not match pattern ".*_test.py"
```

**Comment by goern** [comment]
```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

setup.py:1:2: N812 lowercase 'test' imported as non lowercase 'TestCommand'
setup.py:40:9: F821 undefined name 'sys'
tests/test_unresolved_package_handler.py:20:1: F401 'pytest' imported but unused
tests/test_unresolved_package_handler.py:25:1: F401 'thoth.unresolved_package_handler.unresolved_package_handler.parse_unresolved_package_message' imported but unused
tests/test_unresolved_package_handler.py:37:16: F632 use ==/!= to compare str, bytes, and int literals
```

**Comment by harshad16** [comment]
> One small comment, otherwise LGTM! Why is it not happening for advise-reporter, slo-reporter which does not have setuptools setup? Do we need to add them also there? Thanks!!

@pacospace In advise-reporter and slo-reporter, we aren't using `setup.py`, when the setup.py is available in the application, s2i build process uses it as a source, to get information for APP_MODULE. 
we don't need to add setup.py to other repositories, if they don't require it, however, if we are adding we should include the base setuptools.setup while using `setup.py`

**Comment by harshad16** [comment]
> and ...
> 
> ```
> Tests should end in _test.py.............................................Failed
> - hook id: name-tests-test
> - exit code: 1
> 
> tests/test_unresolved_package_handler.py does not match pattern ".*_test.py"
> ```

@goern , i resolved pre-commit pointed issue for `setup.py` , didn't make the change for other files like `test`.
as i see there is already another PR with pre-commit fixes.

**Comment by harshad16** [comment]
The `setup.py` is removed from the repo as this not a package module.
@pacospace  please have a look

**Comment by fridex** [comment]
Also, this component could be eventually renamed (something like unknown-package-handler). I think it will schedule workflow(s) for more than just to solve a package.

**Pull request merged**